src/Features/Core/Portable/Completion/Providers/TypeImportCompletionItem.cs

```diff
@@ -0,0 +1,113 @@
+        private static readonly string[] s_IntegerOneToNine = { "1", "2", "3", "4", "5", "6", "7", "8", "9" };
+        private static readonly string[] s_aritySuffixesOneToNine = { "`1", "`2", "`3", "`4", "`5", "`6", "`7", "`8", "`9" };
+
+        private const string ContainingNamespaceName = nameof(ContainingNamespaceName);
+        private const string TypeArityName = nameof(TypeArityName);
+
+        public static CompletionItem Create(INamedTypeSymbol typeSymbol, string containingNamespace)
+        {
+            var builder = PooledDictionary<string, string>.GetInstance();
+            builder.Add(ContainingNamespaceName, containingNamespace);
+
+            if (typeSymbol.Arity > 0)
+            {
+                builder.Add(TypeArityName, GetInteger(typeSymbol.Arity));
+            }
+
+            // TODO: Suffix should be language specific, i.e. `(Of ...)` if triggered from VB.
+            return CompletionItem.CreateInternal(
+                 displayText: typeSymbol.Name,
+                 filterText: typeSymbol.Name,
+                 sortText: typeSymbol.Name,
+                 properties: builder.ToImmutableDictionaryAndFree(),
+                 tags: GlyphTags.GetTags(typeSymbol.GetGlyph()),
+                 rules: CompletionItemRules.Default,
+                 displayTextPrefix: null,
+                 displayTextSuffix: typeSymbol.Arity == 0 ? null : "<>",
+                 inlineDescription: containingNamespace,
+                 useEditorCompletionItemCache: true);
+        }
+
+        public static string GetContainingNamespace(CompletionItem item)
+        {
+            return item.Properties.TryGetValue(ContainingNamespaceName, out var containingNamespace)
+                ? containingNamespace
+                : null;
+        }
+
+        public static async Task<CompletionDescription> GetCompletionDescriptionAsync(Document document, CompletionItem item, CancellationToken cancellationToken)
+        {
+            var metadataName = GetMetadataName(item);
+            if (!string.IsNullOrEmpty(metadataName))
+            {
+                var compilation = await document.Project.GetCompilationAsync(cancellationToken).ConfigureAwait(false);
+                var symbol = compilation.GetTypeByMetadataName(metadataName);
+                if (symbol != null)
+                {
+                    var semanticModel = await document.GetSemanticModelAsync(cancellationToken).ConfigureAwait(false);
+
+                    // We choose not to display the number of "type overloads" for simplicity. 
+                    // Otherwise, we need additonal logic to track internal and public visible
+                    // types separately, and cache both completion items.
+                    return await CommonCompletionUtilities.CreateDescriptionAsync(
+                        document.Project.Solution.Workspace,
+                        semanticModel,
+                        position: 0,
```

Comment:
what is this position for?

Reply:
Maybe @dpoeschl or @jasonmalinowski konws?

Reply:
so, looks like position you need to save it in completion item when you create in property from completioncontext so, when description create string for symbol, it uses right qualifier, but your case I guess it will be always fully qualified name?